spec/setup/helpers/abs_helper_spec.rb

```diff
@@ -1081,23 +1109,176 @@ class AbsHelperClass
 
     context "when an invalid resource host array is specified" do
       it "reports the error and returns false" do
-        hosts = TEST_INVALID_ABS_RESOURCE_HOSTS
-        message = "The specified resource host array is not valid: #{hosts}"
-        allow(subject).to receive(:puts)
+        json_hosts = TEST_INVALID_ABS_RESOURCE_HOSTS
+        message = /Invalid abs_resource_hosts JSON specified/
+
+        # parse_abs_resource_hosts should raise (this method depends on this behavior)
+        expect(subject).to receive(:parse_abs_resource_hosts)
+          .with(json_hosts).and_raise(RuntimeError, message)
+
+        # rescue and report the error message
         expect(subject).to receive(:puts).with(message)
 
-        expect(subject.valid_abs_resource_hosts?(hosts)).to eq(false)
+        expect(subject.valid_abs_resource_hosts?(json_hosts)).to eq(false)
       end
     end
 
-    context "when an nil is specified" do
+    context "when invalid JSON is specified" do
       it "reports the error and returns false" do
-        allow(subject).to receive(:puts)
-        expect(subject).to receive(:puts).with("A valid hosts array is required; nil was specified")
+        json_hosts = "THIS IS NOT JSON!"
+        message = /encountered parsing the hosts array/
+
+        # parse_abs_resource_hosts should raise (this method depends on this behavior)
+        expect(subject).to receive(:parse_abs_resource_hosts)
+          .with(json_hosts).and_raise(RuntimeError, message)
+
+        # rescue and report the error message
+        expect(subject).to receive(:puts).with(message)
+
+        expect(subject.valid_abs_resource_hosts?(json_hosts)).to eq(false)
+      end
+    end
+
+    context "when nil is specified" do
+      it "reports the error and returns false" do
+        message = /encountered parsing the hosts array/
+        expect(subject).to receive(:puts).with(message)
 
         expect(subject.valid_abs_resource_hosts?(nil)).to eq(false)
       end
     end
   end
+
+  describe "#parse_abs_resource_hosts" do
+    before do
+      allow(subject).to receive(:puts)
+    end
+
+    context "when a valid hosts JSON string is specified" do
```

Comment:
FYI, the simplest valid json which is also an invalid hosts string (`'{}'`) is not covered and will not throw the expected error.

Reply:
Thanks; updated to raise an error and added an example to the spec.